public/js/index.js

```diff
@@ -0,0 +1,274 @@
+let subredditArray = [
+  [
+    "https://www.reddit.com/r/tifu.json",
+    "https://www.reddit.com/r/HighQualityGifs.json",
+    "https://www.reddit.com/r/science.json",
+    "https://www.reddit.com/r/gaming.json",
+    "https://www.reddit.com/r/Showerthoughts.json",
+    "https://www.reddit.com/r/MechanicalKeyboards.json",
+    "https://www.reddit.com/r/NoStupidQuestions.json",
+    "https://www.reddit.com/r/nonononoyes.json"
+  ],
+  "https://www.reddit.com/r/oddlysatisfying.json",
+  "https://www.reddit.com/r/catloaf.json",
+  "https://www.reddit.com/r/peoplefuckingdying.json"
+];
+
+//log down which subreddit was clicked
+let currentSubreddit;
+let redditJSONLimitAfter;
+//current date in seconds
+var date = new Date().getTime() / 1000;
+
+//navbar
+let findButtons = document.querySelectorAll(".subreddits");
+//Post area
+let postArea = document.getElementById("postArea");
+
+//Click navbar to load
+for (let i = 1; i < findButtons.length; i++) {
+  findButtons[i].addEventListener("click", grabData);
+  function grabData() {
+    postArea.innerHTML = "";
+    let subreddits = new XMLHttpRequest();
+    subreddits.open("GET", subredditArray[i]);
+    subreddits.send();
+    subreddits.addEventListener("load", loadData);
+  }
+}
+
+//Scroll down load
+setTimeout(window.addEventListener("scroll", loadMore), 1000);
+
+function loadMore() {
+  if (window.scrollY + window.innerHeight >= document.body.offsetHeight) {
+    const subreddits = new XMLHttpRequest();
+    subreddits.addEventListener("load", loadData);
+    subreddits.open(
+      "GET",
+      currentSubreddit + "?limit=10&after=" + redditJSONLimitAfter
+    );
+    subreddits.send();
+  }
+}
+
+//Populate random posts
+function generateRandom() {
+  postArea.innerHTML = "";
+  let randomSubreddits = new XMLHttpRequest();
+  randomSubreddits.open(
+    "GET",
+    subredditArray[0][Math.floor(Math.random() * subredditArray[0].length)]
+  );
+  randomSubreddits.send();
+  randomSubreddits.addEventListener("load", loadData);
+}
+generateRandom();
```

Comment:
If all the buttons are tied to directly, then you won't need a separate generateRandom function. It would just be part of the function that's bound to the random button.